features/activity/retry_on_error/feature.ts

```diff
@@ -0,0 +1,25 @@
+import { Context } from '@temporalio/activity';
+import { WorkflowFailedError } from '@temporalio/client';
+import { Feature } from '@temporalio/harness';
+import * as assert from 'assert';
+import { workflow } from './feature.workflow';
+
+export const feature = new Feature({
+  workflow,
+  activities: {
+    alwaysFailActivity,
+  },
+  checkResult: async (runner, handle) => {
+    try {
+      await runner.waitForRunResult(handle);
+      throw new Error('expected failure');
+    } catch (err) {
+      assert.ok(err instanceof WorkflowFailedError);
+      assert.equal(err.cause?.cause?.message, 'activity attempt 5 failed');
```

Comment:
Can / should we assert the causes are the right instances? (ActivitiyFailure -> ApplicationFailure)

Reply:
I specifically called this out as a best practice in the README which unfortunately had to be put in a separate PR. From https://github.com/temporalio/sdk-features/pull/2/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5:

> No need to over-assert on a bunch of values, just confirm that the feature does what is expected via its output.

Could you imagine the sheer number of assertions I could perform to make sure everything was set right on error and happened right on error?

Reply:
I was just verifying, I have read the best practices and was wondering if this would violate them.

Reply:
:+1: There's no right or wrong in this case IMO.